src/app/games/diam/tests/DiamRules.spec.ts

```diff
@@ -0,0 +1,224 @@
+                [__, __, __, __, __, __, __, __],
+                [B1, __, __, __, __, __, __, B2],
+                [A1, __, __, __, __, __, __, A2],
+            ], 4);
+            // when moving the stack starting at A2 clockwise
+            const move: DiamMove = DiamMoveShift.fromRepresentation(new Coord(7, 3), 'clockwise');
+            // then the move suceeds and the stack is moved on top of the other one
+            const expectedState: DiamState = DiamState.fromRepresentation([
+                [B2, __, __, __, __, __, __, __],
+                [A2, __, __, __, __, __, __, __],
+                [B1, __, __, __, __, __, __, __],
+                [A1, __, __, __, __, __, __, __],
+            ], 5);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+        });
+        it('should allow moving from the middle of a stack', () => {
+            // given a state where a shift can be made
+            const state: DiamState = DiamState.fromRepresentation([
+                [__, __, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __, __],
+                [A1, __, __, __, __, __, __, B2],
+                [B1, __, __, __, __, __, __, A2],
+            ], 4);
+            // when moving the stack starting at A1 counterclockwise
+            const move: DiamMove = DiamMoveShift.fromRepresentation(new Coord(0, 2), 'counterclockwise');
+            // then the move is legal
+            const expectedState: DiamState = DiamState.fromRepresentation([
+                [__, __, __, __, __, __, __, __],
+                [__, __, __, __, __, __, __, A1],
+                [__, __, __, __, __, __, __, B2],
+                [B1, __, __, __, __, __, __, A2],
+            ], 5);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+        });
+        it('should allow moving a full stack', () => {
+            // given a state where a shift can be made
+            const state: DiamState = DiamState.fromRepresentation([
+                [B2, __, __, __, __, __, __, __],
+                [A2, __, __, __, __, __, __, __],
+                [B1, __, __, __, __, __, __, __],
+                [A1, __, __, __, __, __, __, __],
+            ], 4);
+            // when moving the stack starting at A1 counterclockwise
+            const move: DiamMove = DiamMoveShift.fromRepresentation(new Coord(0, 3), 'counterclockwise');
+            // then the move is legal
+            const expectedState: DiamState = DiamState.fromRepresentation([
+                [__, __, __, __, __, __, __, B2],
+                [__, __, __, __, __, __, __, A2],
+                [__, __, __, __, __, __, __, B1],
+                [__, __, __, __, __, __, __, A1],
+            ], 5);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+        });
+        it('should forbid moving a stack if the first piece is not owned by the player', () => {
```

Comment:
should forbid moving a sub-stack if its lowest piece is not owner by the player :p ?

Reply:
fait